projects/aws/bottlerocket-bootstrap/pkg/kubeadm/utils.go

```diff
@@ -170,3 +175,67 @@ func unmarshalIntoMap(content []byte) (map[string]interface{}, error) {
 
 	return parsedMap, err
 }
+
+type EbsInitControl struct {
+	Cancel  context.CancelFunc
+	OkChan  chan bool
+	Timeout <-chan time.Time
+}
+
+// startEbsInit starts the ebs-init goroutine in the background
+// if marker file is present. Currenlty ebs-init is best-effort
+// and if it fails it won't prevent the instance from bootstrapping.
+// A 2 minutes timeout is added so that instance bootstrap time is
+// not inflated.
+func startEbsInit() *EbsInitControl {
+	if _, err := os.Stat(ebsInitMarker); err == nil {
+		okChan := make(chan bool)
+		ctx, cancel := context.WithCancel(context.Background())
+		fmt.Printf("Starting ebs-init \n")
+		ebsInitControl := &EbsInitControl{}
+		ebsInitControl.Timeout = time.After(2 * time.Minute)
+		ebsInitControl.Cancel = cancel
+		ebsInitControl.OkChan = okChan
```

Comment:
```suggestion
		ebsInitControl := &EbsInitControl{
		    Timeout = time.After(2 * time.Minute)
		    Cancel = cancel
		    OkChan = okChan
		}
```

Reply:
> Looks good, few nits. I also want to make sure while we do not need etcdadm changes right now for your current usecase, you might need to add the same calls to etcadm flows as well for ebs initialization.

Do you want me to make the changes now?

Reply:
No need to make changes now, just a headup that you might wanna do it when you work with etcdadm. The tests are failing with checksum mismatch since we are changing the code. You should be able to get the checksums from the test and add them if they look alright.

Reply:
Got it, will adjust the checksums in another commit then.